### PR TITLE
support getting node and core number in Python

### PR DESCRIPTION
**Pull request by yangw1234**
## What changes were proposed in this pull request?

support getting node and core number in Python

## How was this patch tested?

manually tested

### CODE REVIEW

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/python/api/PythonBigDL.scala [hunk 1]
```diff
@@ -2338,6 +2338,10 @@ class PythonBigDL[T: ClassTag](implicit ev: TensorNumeric[T]) extends Serializab
     Engine.init
   }
 
+  def getNodeAndCoreNumber(): Array[Int] = {
```
Comment (hkvision) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/python/api/PythonBigDL.scala]: JList[Int]? Will Array[Int] work?

File: pyspark/bigdl/util/common.py [hunk 2]
```diff
@@ -446,6 +446,9 @@ def init_engine(bigdl_type="float"):
 def init_executor_gateway(sc, bigdl_type="float"):
     callBigDlFunc(bigdl_type, "initExecutorGateway", sc, sc._gateway._gateway_client.port)
 
+def get_node_and_core_number(bigdl_type="float"):
+    result = callBigDlFunc(bigdl_type, "getNodeAndCoreNumber")
```
Comment (hkvision) [inline pyspark/bigdl/util/common.py]: Add a unit test?

Comment (yangw1234) [reply pyspark/bigdl/util/common.py]: done

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/python/api/PythonBigDL.scala (hunk 1 above)
Comment (yangw1234) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/python/api/PythonBigDL.scala]: yes, it works

**Comment by yangw1234** [comment]
pr validation pass

**Pull request merged**